### DISCUSSION TITLE
Bug? WeeklyPartitionsDefinition and context.asset_partition_key_range_for_output()

### DISCUSSION THREAD

**Discussion by clayheaton** [dagster-io/dagster#17575] [Q&A] [answered]
I'm trying to set up weekly partitioning on a test asset. Here's how it looks:

```python
@asset(
    # ...
    partitions_def=WeeklyPartitionsDefinition(start_date="2023-09-01", day_offset=1),
    metadata={
        "partition_expr": "WEEK_START_DATE",
    },
    # ...
)
def TEST_WEEKLY_PARTITIONS(context: AssetExecutionContext) -> pd.DataFrame:
    """
    Simple test of WeeklyPartitionDefinition.
    """
    range = context.asset_partition_key_range_for_output()
    context.log.info("context.asset_partition_key_range_for_output()")
    context.log.info(range)
    context.log.info("context.asset_partition_keys_for_output()")
    context.log.info(context.asset_partition_keys_for_output())
    start_date = range.start
    end_date = range.end

    context.log.info(
        "Testing WeeklyPartitionsDefinition with {start_date} to {end_date}".format(
            start_date=start_date, end_date=end_date
        )
    )

    # ...
    return df
```

This should be a simple set up, with the only difference being I want the weeks to start on Monday, hence the `day_offset=1`.

The issue is that the range returned is just a single day. Here's how the logging appears:

![Screenshot 2023-11-01 at 11 41 50 AM](https://github.com/dagster-io/dagster/assets/205364/3425e3b0-407d-4be6-9fde-5fa290ac1b23)

Is this a bug or am I incorrectly fetching the partition range?

**Comment by sryza** [comment, marked as answer]
Hey @clayheaton - a "partition key range" refers to range of partition keys, closed at the beginning and open at the end. So `PartitionKeyRange(start="2023-09-04", end="2023-09-04")` contains a single partition: "2023-09-04".

That partition corresponds to a time window. In this case, because it's a `WeeklyPartitionsDefinition`, the time window is midnight on 2023-09-04 to midnight on 2023-09-11.

To get the time window, you can use `context.partition_time_window`.
